src/parsers/perf/perfparser.cpp

```diff
@@ -838,6 +841,24 @@ class PerfParserPrivate : public QObject
             }
         }
 
+        const auto costAggregation = Settings::instance()->costAggregation();
+
+        if (costAggregation == Settings::CostAggregation::ByThread) {
+            for (auto& thread : bottomUpResult.root.children) {
+                if (tidToName.contains(thread.symbol.symbol)) {
```

Comment:
this is a classic performance antipattern - you do two hash lookups here which is wasteful. instead, use find + iterators:

```
auto it = tidToName.constFind(thread.symbol.symbol);
if (it != tidToName.constEnd()) {
    thread.symbol.prettySymbol = *it;
    thread.symbol.symbol = *it;
}
```